### ISSUE TITLE
Align declarations inserts an unexpected number of spaces

### ISSUE THREAD

**Issue by fabianlupa** [SAP/abap-cleaner#14]
Before

```abap
    TYPES: BEGIN OF al11_file,
             dirname  TYPE dirname_al11,
             name     TYPE filename_al11,
             type     TYPE c  LENGTH 10,
             len      TYPE p  LENGTH 8 DECIMALS 0,
             owner    TYPE fileowner_al11,
             mtime    TYPE p LENGTH 6 DECIMALS 0,
             mode     TYPE c LENGTH 9,
             useable  TYPE c LENGTH 1,
             subrc    TYPE c LENGTH 4,
             errno    TYPE c LENGTH 3,
             errmsg   TYPE c LENGTH 40,
             mod_date TYPE d,
             mod_time TYPE c LENGTH 8,
             seen     TYPE c LENGTH 1,
             changed  TYPE c LENGTH 1,
             status   TYPE c LENGTH 1,
           END OF al11_file,
           al11_file_tab TYPE STANDARD TABLE OF al11_file.
    FIELD-SYMBOLS <file_list> TYPE al11_file_tab.
```

After

```abap
    TYPES: BEGIN OF al11_file,
             dirname     TYPE dirname_al11,
             name        TYPE filename_al11,
             type        TYPE c                           LENGTH 10,
             len         TYPE p                           LENGTH 8 DECIMALS 0,
             owner       TYPE fileowner_al11,
             mtime       TYPE p                           LENGTH 6 DECIMALS 0,
             mode        TYPE c                           LENGTH 9,
             useable     TYPE c                           LENGTH 1,
             subrc       TYPE c                           LENGTH 4,
             errno       TYPE c                           LENGTH 3,
             errmsg      TYPE c                           LENGTH 40,
             mod_date    TYPE d,
             mod_time    TYPE c                           LENGTH 8,
             seen        TYPE c                           LENGTH 1,
             changed     TYPE c                           LENGTH 1,
             status      TYPE c                           LENGTH 1,
           END OF al11_file,
           al11_file_tab TYPE STANDARD TABLE OF al11_file.
    FIELD-SYMBOLS <file_list> TYPE al11_file_tab.
```

**Comment by fabianlupa** [comment]
I found some other examples. It seems like the LENGTH addition is aligned after the type of the longest definition in the chain, therefore above STANDARD TABLE OF al11_file. Which I guess is fine if you know about it. Or the component in the type statement that is not part of the BEGIN OF / END OF could have been extracted to its own statement. Here's a more reasonable example of the same:

```abap
  TYPES:
    access_type           TYPE c          LENGTH 1,
    mode                  TYPE c          LENGTH 15,
    encoding              TYPE c          LENGTH 15,
    linefeed              TYPE c          LENGTH 10,
    endian                TYPE c          LENGTH 1,
    BEGIN OF mode_options,
      encoding            TYPE encoding,
      linefeed            TYPE linefeed,
      endian              TYPE endian,
      codepage            TYPE cpcodepage,
    END OF mode_options,
    BEGIN OF os_additions,
      type                TYPE string,
      filter              TYPE string,
    END OF os_additions,
    replacement_character TYPE c          LENGTH 1.
```

**Comment by fabianlupa** [comment]
Also doesn't this directly implement the anti pattern listed in the Clean ABAP rule?

https://github.com/SAP/styleguides/blob/main/clean-abap/CleanABAP.md#dont-align-type-clauses

Independant statements are also aligned which is the exact example of what you are not supposed to do?

```abap
    DATA file_system      TYPE REF TO zabc_file_system.
    DATA path             TYPE REF TO zabc_path.
    DATA current_codepage TYPE cpcodepage.
```

**Comment by jmgrassau** [comment]
Hi Fabian, 

thanks a lot for these examples, this should also be fixed in version 1.2! Now, with default settings in "Align declarations", I get (original code -> cleaned code):

![image](https://github.com/SAP/abap-cleaner/assets/79639298/6e54b484-2611-4f57-8e57-b20752b4f852)

![image](https://github.com/SAP/abap-cleaner/assets/79639298/21d0d995-a464-46c5-80c5-4e53a40d5d65)

> Also doesn't this directly implement the anti pattern listed in the Clean ABAP rule?

Yes, it indeed does – that's why the "Align declarations" rule is deactivated in the "essential" Profile (in which only those rules are active which are explicitly demanded by the style guide). 

However, from my experience, this is a rather controversial rule in the style guide, so ABAP cleaner offers "Align declarations" to whoever likes it. And you could argue: If nobody wanted to align TYPEs, then why does Pretty Printer do it on chains, structures and METHODS declarations? Why does the style guide itself have many positive examples with aligned TYPEs? …

Kind regards,
Jörg-Michael

**Comment by fabianlupa** [comment]
Thanks! My initial understanding of the tool was that the default settings do implement the rules of the Clean ABAP styleguide. After reading https://github.com/SAP/abap-cleaner/blob/main/docs/profiles.md I see the non-default Essential profile does that instead. I also found out what the "<->" means in the configuration dialog by reading that.

I do wonder though why this isn't the other way around as at least according to the guide itself it is the [official reference guide for S/4HANA development](https://github.com/SAP/styleguides/commit/012d2e8bdc19de321ed51c1a2310dd07e4f87de3), therefore I would have assumed most users want those settings.

Regarding the rule itself I have no opinion, especially when there's no manual work involved in following it. I just find the "alignment" / consistency between the tools (Code Pal, ADT, ABAP Cleaner) and guidelines (Clean ABAP GitHub, Clean ABAP Book) to be very important for adoption without frustration.
